frontend/appflowy_flutter/lib/plugins/inline_actions/handlers/inline_page_reference.dart

```diff
@@ -125,7 +125,15 @@ class InlinePageReferenceService extends InlineActionsDelegate {
 
       items = allViews
           .where(
-            (view) => view.name.toLowerCase().contains(search.toLowerCase()),
+            (view) =>
+                view.id != currentViewId &&
+                    view.name.toLowerCase().contains(search.toLowerCase()) ||
+                (view.name.isEmpty && search.isEmpty) ||
+                (view.name.isEmpty &&
+                    LocaleKeys.menuAppHeader_defaultNewPageName
+                        .tr()
+                        .toLowerCase()
+                        .contains(search.toLowerCase())),
```

Comment:
The filter logic looks a bit complicated. How about abstracting a function and adding some comments?